go-controller/pkg/node/gateway_localnet.go

```diff
@@ -119,11 +119,7 @@ func (l *localPortWatcher) AddService(svc *kapi.Service) {
 }
 
 func (l *localPortWatcher) UpdateService(old, new *kapi.Service) {
-	if reflect.DeepEqual(new.Spec.Ports, old.Spec.Ports) &&
-		reflect.DeepEqual(new.Spec.ExternalIPs, old.Spec.ExternalIPs) &&
-		reflect.DeepEqual(new.Spec.ClusterIP, old.Spec.ClusterIP) &&
-		reflect.DeepEqual(new.Spec.Type, old.Spec.Type) &&
-		reflect.DeepEqual(new.Status.LoadBalancer.Ingress, old.Status.LoadBalancer.Ingress) {
+	if apiequality.Semantic.DeepEqual(new.Spec, old.Spec) && apiequality.Semantic.DeepEqual(new.Status, old.Status) {
```

Comment:
are you sure you want to do this? I thought we were only checking parts of a service we care about. This seems like it could be too generic...

Reply:
it was more a exploratory thing, the if was growing too much :)

Reply:
will keep it as it was